app/apps/server/communication/rest.js

```diff
@@ -516,6 +516,33 @@ export class AppsRestApi {
 			},
 		});
 
+		this.api.addRoute('icon/:id', { authRequired: false }, {
```

Comment:
I know I can be hated for this because there is another endpoint `:id/icon` already: https://github.com/RocketChat/Rocket.Chat/pull/16522/files#diff-1d7b87614deb7cd95b3cfaa7420842a7R506

Reply:
IIRC the other one is not used anywhere? 🤔 I think I've searched for it before and haven't found anything. If that's the case, you can just change the behavior of the other one 😬

Reply:
Since we're targeting a major version, I could introduce this breaking change.